src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -85,15 +82,9 @@ public void slowOff() {
     isSlow = false;
   }
 
-  public void steppingSpeedUp() {
-    steppingSpeedUp = true;
-  }
-
-  public void steppingSpeedDown() {
-    steppingSpeedDown = true;
-  }
-
   private double adjustSpeed(double speed) {
+    SmartDashboard.putString("Is Turbo", isTurbo ? "yes" : "no");
+    SmartDashboard.putString("Is Slow", isSlow ? "yes" : "no");
     if (isTurbo) {
```

Comment:
I think we can make our lives a lot easier here when we're trying to deal with `slow` and `turbo`.

We can probably do similar to https://docs.wpilib.org/en/latest/docs/software/commandbased/convenience-features.html in `RobotContainer.java`:

```java
  // turbo
  driverRightTrigger
    .whenPressed(() -> driveSubsystem.setMaxSpeed(DriveConstants.TURBO_SPEED))
    .whenReleased(() -> driveSubsystem.setMaxSpeed(DriveConstants.NORMAL_SPEED));

  // slow
  driverLeftTrigger
    .whenPressed(() -> driveSubsystem.setMaxSpeed(DriveConstants.SLOW_SPEED))
    .whenReleased(() -> driveSubsystem.setMaxSpeed(DriveConstants.NORMAL_SPEED));
```

Then in adjust speed we're just using some (new) variable `maxSpeed`:

```java
if (speed >= 0) {
  return Math.min(speed, maxSpeed)
} else {
  return Math.max(speed, maxspeed)
}
```

This way, we're just changing the potential `maxSpeed` with the triggers and not trying to remember what `mode` we're in.  So we simply have to know if we're dealing with `+` or `-` speeds to choose `.min` or `.max`

If we want to stick with our current paradigm and call `turboOn` then `DriveSubsystem#turboOn()` would just be:

```java
public void turboOn() {
  maxSpeed = DriveConstants.TURBO_SPEED;
}
```

Similarly for the other commands (`TurboOff`, `SlowOn`, `SlowOff`)

Reply:
Can we work on this when you get here?